src/components/ha-form/ha-form.ts

```diff
@@ -136,7 +137,7 @@ export class HaForm extends LitElement implements HaFormElement {
         display: block;
       }
       .root > *:not([own-margin]) {
-        margin-bottom: 24px;
+        margin-bottom: 16px;
```

Comment:
I am pretty sure that this was aligned with the "own-margin" components margin below. We should test this better.

Reply:
I think it was driven by the helper text of mwc-textfield that hides when not in focus but still takes up space.